osquery/tables/system/windows/processes.cpp

```diff
@@ -36,85 +37,83 @@ std::set<long> getProcList(const QueryContext& context) {
         pidlist.insert(pid);
       }
     }
-    return pidlist;
-  } else {
-    WmiRequest request("SELECT ProcessId FROM Win32_Process");
-    if (request.getStatus().ok()) {
-      long placeHolder;
-      Status s;
-      for (auto const& result : request.results()) {
-        s = result.GetLong("ProcessId", placeHolder);
-        if (!s.ok()) {
-          continue;
-        }
-        pidlist.insert(placeHolder);
-      }
-    }
-    return pidlist;
   }
-}
 
-void genProcess(long pid, QueryData& results_data) {
-  std::stringstream ss;
-  ss << "SELECT * FROM Win32_Process WHERE ProcessId=" << pid;
+  /// If there are no constraints, pidlist will be an empty set
+  return pidlist;
+}
 
-  WmiRequest request(ss.str());
-  if (request.getStatus().ok()) {
-    std::vector<WmiResultItem>& results = request.results();
-    if (results.size() == 1) {
-      Row r;
-      Status s;
-      long lPlaceHolder;
-      std::string sPlaceHolder;
-
-      s = results[0].GetLong("ProcessId", lPlaceHolder);
-      r["pid"] = s.ok() ? BIGINT(lPlaceHolder) : BIGINT(-1);
-      s = results[0].GetString("Name", sPlaceHolder);
-      r["name"] = SQL_TEXT(sPlaceHolder);
-      s = results[0].GetString("ExecutablePath", sPlaceHolder);
-      r["path"] = SQL_TEXT(sPlaceHolder);
-      s = results[0].GetString("CommandLine", sPlaceHolder);
-      r["cmdline"] = SQL_TEXT(sPlaceHolder);
-      s = results[0].GetString("ExecutionState", sPlaceHolder);
-      r["state"] = SQL_TEXT(sPlaceHolder);
-      s = results[0].GetLong("ParentProcessId", lPlaceHolder);
-      r["parent"] = BIGINT(lPlaceHolder);
-      s = results[0].GetLong("Priority", lPlaceHolder);
-      r["nice"] = INTEGER(lPlaceHolder);
-      r["on_disk"] = osquery::pathExists(r["path"]).toString();
-
-      // TODO: some of these such as cwd, wired_size, phys_footprint
-      // should be retrievable either via Windows API or WMI
-      r["cwd"] = "";
-      r["root"] = "";
-
-      r["pgroup"] = "-1";
-      r["uid"] = "-1";
-      r["euid"] = "-1";
-      r["suid"] = "-1";
-      r["gid"] = "-1";
-      r["egid"] = "-1";
-      r["sgid"] = "-1";
-
-      r["wired_size"] = "0";
-      r["resident_size"] = "0"; // Populate with WorkingSetSize (VT_BSTR)
-      r["phys_footprint"] = "0";
-
-      r["user_time"] = "0";
-      r["system_time"] = "0";
-      r["start_time"] = "0";
-
-      results_data.push_back(r);
-    }
-  }
+void genProcess(const WmiResultItem& result, QueryData& results_data) {
+  Row r;
+  Status s;
+  long lPlaceHolder;
+  std::string sPlaceHolder;
+
+  s = result.GetLong("ProcessId", lPlaceHolder);
+  r["pid"] = s.ok() ? BIGINT(lPlaceHolder) : BIGINT(-1);
+  s = result.GetString("Name", sPlaceHolder);
```

Comment:
For the second parameter, just use r["name"]. Ditch the s = in all of these if you're not using them.